packages/core/src/fhir-deduplication/resources/medication-administration.ts

```diff
@@ -19,35 +44,45 @@ export function deduplicateMedAdmins(medications: MedicationAdministration[]): {
  * Approach:
  * 1 map, where the key is made of:
  * - medicationReference ID
- * - status
  * - date
+ * - dosage
  */
 export function groupSameMedAdmins(medAdmins: MedicationAdministration[]): {
   medAdminsMap: Map<string, MedicationAdministration>;
-  remainingMedAdmins: MedicationAdministration[];
   refReplacementMap: Map<string, string[]>;
 } {
   const medAdminsMap = new Map<string, MedicationAdministration>();
   const refReplacementMap = new Map<string, string[]>();
-  const remainingMedAdmins: MedicationAdministration[] = [];
+
+  function assignMostDescriptiveStatus(
+    master: MedicationAdministration,
+    existing: MedicationAdministration,
+    target: MedicationAdministration
+  ): MedicationAdministration {
+    master.status = pickMostDescriptiveStatus(statusRanking, existing.status, target.status);
```

Comment:
is this part tested? usually this doesnt actually update and you need to create a new variable with a spread operator?

Reply:
ditto for the 2 other places where its used

Reply:
nevermind u tested this